Topics/starting-a-data-project.md

```diff
@@ -0,0 +1,254 @@
+
+_Any existing software project that is not under version control is considered a dead project, and the responsible developers are considered crazy._
+
+Data Science projects (which make heavy use of software) are no different, indeed! 
+
+They also have the _additional problem of data versioning_, which is the raw material on which you work most. 
+
+Always having the versions of the data, from raw (just collected) to clean, keeping every intermediate processing phase, is perhaps the most important [best practice](https://medium.com/thelaunchpad/retracing-your-steps-in-machine-learning-ml-versioning-74d19a66bd08) when doing a Data Science project.
+
+[**Here**](https://rogerdudler.github.io/git-guide/index.html) you can find a simple guide to Git. Learn it, it's freaking worth (and necessary).
+
+Documenting your work with Git is crucial: read [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
+
+**Must read:**
+[**Ten Simple Rules for Taking Advantage of Git and GitHub**](http://dx.plos.org/10.1371/journal.pcbi.1004947)
+
+### Documentation
+
+Like any project, documenting the work done is fundamental to the success of the project. 
+
+We don't need to list the benefits that good documentation brings to a project, so we immediately understand what are the best practices to keep in mind when we produce documentation for our projects.
+
+If you still want to learn more, read [this article](https://towardsdatascience.com/why-you-should-document-your-work-as-a-data-scientist-a265af8a373?gi=bc5bae43230e) and [this other](https://hackernoon.com/why-you-should-document-your-self-documenting-code-1105a8a6852e).
+
+[**This guide**](https://managing-qualitative-data.org/modules/2/a/) explains in detail how to document data collection and its organization.
+
+Read also [Ten simple rules for documenting scientific software](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6301674/).
+
+You can choose among different ways to document your project, but Virgilio recommends you to use [Sphinx](http://www.sphinx-doc.org/en/master/), the official Python automated docs library. 
+
+Remember that documenting your code and project steps it's **NEVER wasted time**.
+
+**Must read:**
+[**Ten simple rules for documenting scientific software**](http://dx.plos.org/10.1371/journal.pcbi.1006561)
+
+
+### Conclusions
+
+After reading this guide and the resources it contains, you should be equipped with all the necessary best practices when starting a new Data Science project. 
+
+In the next sections of Purgatorio you will begin to put these practices into practice, and you will see how grateful you are to yourself! 
```

Comment:
> Purgatorio

Terms like this must be hyperlinked - to the page(s) where it's best explained. Is there a glossary or page that explains such terms?

Reply:
which terms?

Reply:
Terms like `Purgatorio` and all Virgilio created terms - and hence I suggested the glossary.

At this point, it appears either you were not fully focussed when reading my comments or the device you used didn't show you my comments clearly. I suggest to use a Desktop version of Github when not on a laptop or desktop computer.

Reply:
I could not see them. BUt I was from mobile, now from desktop it shows

Reply:
ok for now let's leave them anonymous, I think a glossary isnt needed yet

Reply:
At least explain to the reader or give them a link to the meaning of the unknown or new terms IMO

Reply:
It would be fair to the reader to know what Virgilio terms mean atleast in the beginning

Reply:
I think about a glossary, soon it will be needed. But it's not a priority, to understand what is Paradiso, Purgatorio and Inferno is pretty straighforward, and it s in the map and the homepage